services/venice-controller/src/main/java/com/linkedin/venice/controller/systemstore/SystemStoreRepairTask.java

```diff
@@ -182,39 +170,80 @@ void checkAndSendHeartbeatToSystemStores(
               .add(VeniceSystemStoreType.DAVINCI_PUSH_STATUS_STORE.getSystemStoreName(store.getName()));
         }
         if (!store.isStoreMetaSystemStoreEnabled()) {
           newUnhealthySystemStoreSet.add(VeniceSystemStoreType.META_STORE.getSystemStoreName(store.getName()));
         }
-
       }
     }
-
-    for (Store store: getParentAdmin().getAllStores(clusterName)) {
+    // Second pass only check system store and potentially send heartbeat message to validate health.
+    for (Store store: storeList) {
       if (!shouldContinue(clusterName)) {
         return;
       }
+
       // This pass we only scan user system store.
       if (!(VeniceSystemStoreUtils.isSystemStore(store.getName())
           && VeniceSystemStoreUtils.isUserSystemStore(store.getName()))) {
         continue;
       }
+
+      // Skipping migration store.
+      if (store.isMigrating()) {
+        LOGGER.info("Store: {} is being migrated, will skip it for now.", store.getName());
+        continue;
+      }
+
       // We will not check newly created system stores.
       if (isStoreNewlyCreated(userStoreToCreationTimestampMap.getOrDefault(store.getName(), 0L))) {
         continue;
       }
 
+      // Skip checking store that is already added in the unhealthy list in the previous scan.
+      if (newUnhealthySystemStoreSet.contains(store.getName())) {
+        continue;
+      }
+
+      /**
+       * Checking the largest version's age and add stale version to the repair service. Note that in parent region,
+       * there is no current version information, here we use the latest version by the creation time to compute the
+       * version age.
+       */
+      List<Version> storeVersionList = store.getVersions();
+      if (storeVersionList.isEmpty()) {
+        LOGGER.info("Adding the system store: {} to the repair set as there is no version.", store.getName());
+        newUnhealthySystemStoreSet.add(store.getName());
+        continue;
+      }
+      long versionAgeInMs =
+          System.currentTimeMillis() - storeVersionList.stream().mapToLong(Version::getCreatedTime).max().getAsLong();
+      if (versionAgeInMs > TimeUnit.DAYS.toMillis(getVersionRefreshThresholdInDays())) {
+        LOGGER.info(
+            "Adding the system store: {} to the repair set as the version age: {} exceed threshold.",
+            store.getName(),
+            versionAgeInMs);
+        newUnhealthySystemStoreSet.add(store.getName());
+        continue;
+      }
+
       // Send heartbeat to system store in all child regions.
       long currentTimestamp = System.currentTimeMillis();
       sendHeartbeatToSystemStore(clusterName, store.getName(), currentTimestamp);
       systemStoreToHeartbeatTimestampMap.put(store.getName(), currentTimestamp);
 
       // Sleep to throttle heartbeat send rate.
       try {
-        Thread.sleep(PER_SYSTEM_STORE_HEARTBEAT_CHECK_INTERVAL_IN_MS);
+        Thread.sleep(DEFAULT_PER_SYSTEM_STORE_HEARTBEAT_CHECK_INTERVAL_IN_MS);
```

Comment:
use LatencyUtils::sleep or set the following `      Thread.currentThread().interrupt();` in catch block